common-utils/src/common/deployment/__init__.py

```diff
@@ -294,22 +272,26 @@ def _get_schedule_arg(self):
             str: Command line argument to use for setting the deployment's schedule.
         """
         schedule = self.schedule
-        if isinstance(schedule, CronSchedule):
-            return f"--cron '{schedule.cron}'"
-        elif isinstance(schedule, RRuleSchedule):
-            return f"--rrule '{schedule.rrule}'"
-        elif isinstance(schedule, IntervalSchedule):
-            return f"--interval '{schedule.interval.seconds}'"
-        else:
+        # we should only schedule for main flows
+        if not DEPLOYMENT_TYPE == "main":
             return ""
+        else:
+            if isinstance(schedule, CronSchedule):
+                return f"--cron '{schedule.cron}'"
+            elif isinstance(schedule, RRuleSchedule):
+                return f"--rrule '{schedule.rrule}'"
+            elif isinstance(schedule, IntervalSchedule):
+                return f"--interval '{schedule.interval.seconds}'"
+            else:
+                return ""
```

Comment:
schedules should only be added on main deploys.  I supposed it would be better to deploy as disabled...but I could not find that option.  I can submit an issue.